packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -408,7 +408,22 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
             )
           }
           needQueryInjectHelper = true
-          str().overwrite(start, end, `__vite__injectQuery(${url}, 'import')`)
+          if (
+            !/^('.*'|".*"|`.*`)$/.test(url) ||
+            (!isJSRequest(cleanUrl(url.slice(1, -1))) &&
+              !isCSSRequest(url.slice(1, -1)))
+          ) {
+            str().overwrite(start, end, `__vite__injectQuery(${url}, 'import')`)
+          } else {
+            const depModule = await moduleGraph.ensureEntryFromUrl(url)
```

Comment:
I think that this does not work correctly. Url will always contain some variables and therefore, ensureEntryFromUrl will always return the wrong depModule. 
I assume that cannot be solved while building, but has to be resolved at runtime, when the actual string is handed over to __vite__injectQuery. Right now, I have no Idea how I can access the lastHMRTimestamp on runtime. I need to dig into that.

Reply:
The whole hmr seems to be another story... I am already working on it, but this will take some more time. I will open another PR for that.